File: Foundation/Recursion/basic/ques1.cpp
#include<iostream>
using namespace std;
int printDecreasing(int n){
    if(n==0)   return 0;
    cout<<"hi" << n<<endl;
    printDecreasing(n-1);
    cout<<"bye" << n<<endl;
}
int main(){
    int n;
    cin>> n;
    printDecreasing(n);
    return 0;
}

File: Recursion/ques1.java
import java.util.Scanner;
public class ques1{
    public static Scanner scn = new Scanner(System.in);
    public static void printDecreasing(int n){
        if(n==0)   return;
        System.out.println("hi" + n);
        printDecreasing(n-1);
        System.out.println("bye" + n);
    }
    public static void main(String[] args){
        int n = scn.nextInt();
        printDecreasing(n);
    }
}